### PR TITLE
Remove Sphinx Astropy theme

### PR DESCRIPTION
**Pull request by astrofrog**
As discussed in https://github.com/astropy/astropy-helpers/issues/338, I've moved the Astropy theme out into a standalone repository:

https://github.com/astropy/sphinx-astropy-theme

Now if people try and build the docs and the theme is not installed, they will see the following error:

screen shot 2018-01-19 at 18 11 12

Note however the message in blue which is not a warning (and will therefore not trigger issues for people trying to build without warnings) but which should be prominent enough to catch their attention.

At the moment there aren't any releases of sphinx-astropy-theme, but if people are in favor of this PR, I can go ahead and do a release before we merge this (I was thinking of starting versioning the theme starting at 1.0 since we've been using it for a while).

### CODE REVIEW

**Comment by pllim** [comment]
From a package maintainer POV, I would like switch-over instructions; e.g., "If you update to this new version of astropy-helpers, you need to do this, this, and this to get the new theme and get rid of old code."

**Comment by astrofrog** [comment]
I guess the instructions are just 'if you update to this new version of astropy-helpers, you need to install sphinx-astropy-theme' manually :) It's a good point that maybe we should keep a list of things like that to mention in the PR that auto-updates astropy-helpers though?

**Comment by pllim** [comment]
Perhaps an update to the package template is also in order?

**Comment by astrofrog** [comment]
@pllim - yes but let's wait until more work is done on astropy-helpers, because I think we also want to unbundle numpydoc and sphinx-automodapi, which means we really should just have a sphinx-astropy package that users can install instead of all the different separate packages.

**Comment by astrofrog** [comment]
(the tests are failing because I need to install the theme for the CI to run)

File: setup.py [hunk 1]
```diff
@@ -47,5 +47,6 @@
     cmdclass=cmdclass,
     python_requires='>=3.5',
     zip_safe=False,
+    install_requires=['sphinx-astropy-theme'],
```
Comment (Cadair) [inline setup.py]: You don't actually *need* the theme to use helpers... I can understand why you would want to keep it though.

Comment (Cadair) [reply setup.py]: You could make the `conf.py` fall back to the default sphinx `alablaster` theme if the astropy one isn't installed.

Comment (astrofrog) [reply setup.py]: I don't think we should have a fallback as I think for now users won't want the theme to suddenly change, and it's easier to tell users how to install the theme if there is an error.

**Comment by astrofrog** [comment]
So are we ok with an error if the theme isn't installed? I don't think there is a way to auto-install as if the submodule is present I don't think astropy-helpers is actually installed so ``install_requires`` in the helpers isn't honored.

Alternatively we could do as @Cadair suggests and have a fallback theme.

At the end of the day, the only people building the docs are people making contributions, so I think they might be ok with installing a separate package. By the time 3.1 comes around we might be asking them to just install ``sphinx-astropy`` to build the docs anyway (not just the theme).

**Comment by astrofrog** [comment]
Note that we can also wait until sphinx-astropy is in place before considering how to deal with it here. I'm going to be working on that over the weekend (there's no rush to merge)

**Comment by astrofrog** [comment]
Closing in favor of #368

**Pull request closed without merging**